spec/services/supergroup_sections_spec.rb

```diff
@@ -1,26 +1,30 @@
-require "test_helper"
-
 describe SupergroupSections::Sections do
   include SearchApiHelpers
   include TaxonHelpers
 
   let(:taxon_id) { "12345" }
   let(:base_path) { "/base/path" }
   let(:supergroup_sections) { SupergroupSections::Sections.new(taxon_id, base_path) }
+  let(:sections) { supergroup_sections.sections }
 
   describe "#sections" do
-    before(:all) do
-      Supergroups::PolicyAndEngagement.any_instance.stubs(:tagged_content).with(taxon_id).returns(section_tagged_content_list("case_study"))
-      Supergroups::Services.any_instance.stubs(:tagged_content).with(taxon_id).returns(section_tagged_content_list("form"))
-      Supergroups::GuidanceAndRegulation.any_instance.stubs(:tagged_content).with(taxon_id).returns(section_tagged_content_list("guide"))
-      Supergroups::NewsAndCommunications.any_instance.stubs(:tagged_content).with(taxon_id).returns([])
-      Supergroups::Transparency.any_instance.stubs(:tagged_content).with(taxon_id).returns([])
-      Supergroups::ResearchAndStatistics.any_instance.stubs(:tagged_content).with(taxon_id).returns([])
-      @sections = supergroup_sections.sections
+    before do
+      stub_config = {
+        Supergroups::PolicyAndEngagement => section_tagged_content_list("case_study"),
+        Supergroups::Services => section_tagged_content_list("form"),
+        Supergroups::GuidanceAndRegulation => section_tagged_content_list("guide"),
+        Supergroups::NewsAndCommunications => [],
+        Supergroups::Transparency => [],
+        Supergroups::ResearchAndStatistics => [],
+      }
+
+      stub_config.each do |klass, results|
+        allow_any_instance_of(klass).to receive(:tagged_content).and_return(results)
+      end
     end
 
-    it "returns a list of sections with tagged content" do
-      assert 5, @sections.length
+    it "returns a list of supergroup sections" do
+      expect(sections.length).to be 6
```

Comment:
How come it was 5 before but now it's 6?

Reply:
It's counting the number of supergroups of which there are 6. The previous thing was asserting that `5` was not `nil` and the message to display was `@sections.length`. I'm taking this as an error that I'm correcting.

Reply:
(So to be clear, the previous assertion wasn't testing what it thought it was testing)